internal/registry/registry.go

```diff
@@ -102,7 +102,7 @@ func NewRegistry(cfg config.Discovery, dstore datastore.Datastore, disco discove
 		dstore: dstore,
 	}
 
-	count, err := r.loadPersistedProviders()
+	count, err := r.loadPersistedProviders(context.Background())
```

Comment:
should this one bubble up?

Reply:
Probably makes sense, but only to cancel this if a graceful shutdown is requested while the indexer is still starting up.
Done.